Cult magic attacks and attacks on stun mark can now be blocked by shields

HMBGERDO opened the pull request:
<!-- By ticking or leaving ticked the option "Allow edits and access to secrets by maintainers" you give permission for repository maintainers to push changes to your branch without explicitly asking. -->

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!-- Include a small to medium description of what your PR changes. -->
<!-- Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
<!-- If your PR fixes an issue, add "Fixes #1234" somewhere in the PR description. This will automatically close the bug upon PR submission. -->
This PRs adds checks for shields for cult spell attacks and exploding stun mark

## Why It's Good For The Game
<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Blocking cult hands with shield sounds good. Blocking cultists attacking you with cult mark set is good.

## Testing
<!-- How did you test the PR, if at all? -->
Compiled, attacked with various cult weapons, tried throwing spear.

## Changelog
:cl:
fix: Cult weapons and attacks with stun mark on you now can be blocked with shields.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

chuga-git commented on code/game/gamemodes/cult/cult_items.dm:
```diff
@@ -45,11 +45,13 @@
-			S.trigger()
-	..()
+	. = ..()
+	if(IS_CULTIST(target) || . == FALSE)
```
```suggestion
	if(IS_CULTIST(target) || !.)
```
Parent can return `null`, which is not equivalent to FALSE.

DGamerL replied:
`!` is just a more robust check yeah, `== FALSE` should almost never be done

HMBGERDO replied:
`. == FALSE` required because it will be `FALSE` if attack was blocked. Otherwise it would be `null`, thats why i check here for that. I wish that to return something better than `null` and `FALSE` but we have what we have

chuga-git replied:
The only way (hopefully) that the parent `attack()` will return `null` in this case is if the attacker is pacified, which `. == FALSE` will let go through and detonate the mark even without playing an animation or sound.

chuga-git commented on code/game/gamemodes/cult/cult_items.dm:
```diff
@@ -622,6 +629,9 @@
 
 /obj/item/cult_spear/attack(mob/living/M, mob/living/user, def_zone)
 	. = ..()
+	if(. == FALSE)
```
```suggestion
	if(!.)
```
Same here

HMBGERDO commented:
moving to draft until i understand attack code and how to fix it

HMBGERDO commented:
Alright, it seems to work.
I changed some files that could cause changes in other item behaviour

first of all, `/mob/living/carbon/human/attacked_by()` now actually returns `TRUE` if attack was successful instead of returning `null`(if success) or `FALSE`(if unsuccessful, like blocked)

second, in `/obj/item/proc/attack()` i changed check for successful attack to actually use the logic above. So if attack was blocked in `M.attacked_by(src, user, def_zone)`, `melee_attack_chain` will stop and not call `afterattack` proc

so, if another item doesnt have special `attack` code and it has some `afterattack` effect, it may be blocked and `afterattack` part wont be called

lewcc commented on code/_onclick/item_attack.dm:
```diff
@@ -81,7 +81,7 @@
 
 	user.do_attack_animation(M)
-	. = M.attacked_by(src, user, def_zone)
+	. = !M.attacked_by(src, user, def_zone)
```
This is kind of a deep change, have you verified that this doesn't have any other implications with attacks?

HMBGERDO replied:
as i stated above, it only denies `afterattack`, nothing else
from what i've seen, there are not many weapons with afterattack and nothing should be broken

HMBGERDO replied:
no, i did not test everything with afterattack, but i tested attacking with cult dagger and blade against someone parrying and not parrying and it worked as i wanted

lewcc reviewed (approved):
Should be fine, LGTM
